### PR TITLE
Dns the dual stack support(part 2)

### PR DESCRIPTION
**Pull request by xiaoxiang781216**

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Close https://github.com/apache/incubator-nuttx/pull/664 accidentally, recreate again.

**Comment by xiaoxiang781216** [comment]
@yamt The new PR should resolve your concern, please review the change.

File: libs/libc/netdb/lib_gethostbyaddrr.c [hunk 1]
```diff
@@ -61,8 +60,10 @@
 
 struct hostent_info_s
 {
+  int       hi_addrtypes[CONFIG_NETDB_MAX_IPADDR + 1];
+  int       hi_lengths[CONFIG_NETDB_MAX_IPADDR + 1];
```
Comment (yamt) [inline libs/libc/netdb/lib_gethostbyaddrr.c:64]: is "+ 1" necessary?
hi_addrlist needs +1 for terminating NULL. i guess these two don't.

Comment (xiaoxiang781216) [reply libs/libc/netdb/lib_gethostbyaddrr.c:64]: Yes, it isn't necessary, but it make the struct neat and orderly

Comment (xiaoxiang781216) [reply libs/libc/netdb/lib_gethostbyaddrr.c:64]: @yamt do you want to remove +1?

Comment (yamt) [reply libs/libc/netdb/lib_gethostbyaddrr.c:64]: if they are not necessary, yes

Comment (xiaoxiang781216) [reply libs/libc/netdb/lib_gethostbyaddrr.c]: Done.

**Comment by xiaoxiang781216** [comment]
@patacongo the PR has fixed all comments from @yamt , please help review and merge it.

**Comment by patacongo** [comment]
> 
> 
> @patacongo the PR has fixed all comments from @yamt , please help review and merge it. If you have any new suggestion, let me know.

Will do...

I looked through the changes and saw no issues.  But I am not fully up to speed with all of what you are trying to do.  I have noted all of the review comments and dialog, so I am confident that this has had a good review by others.

**Pull request merged**